### PR TITLE
Move ApplicationPowerHandling to ApplicationComponents

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Move ApplicationPowerHandling to ApplicationComponents

## Motivation and context
Split up the monothonic Application.h into pieces to avoid the hot header.

## How has this been tested?
It builds..

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Sits on top of https://github.com/xbmc/xbmc/pull/21933

### CODE REVIEW

**Review by garbear — APPROVED** [review]
My review approach was similar to https://github.com/xbmc/xbmc/pull/21933. I read about 10 screens and everything looked good. Then I hit End and was instantly teleported to the bottom of the page. I can confidently say that I saw no errors.

**Comment by lrusak** [comment]
I'm going to hold off looking at this until #21933 get's merged, then this can be rebased and it will be a single commit to review 😄

**Review by garbear — APPROVED** [review]
It was hard to see that this PR included commits from the other PRs. You can open a PR against another PR branch instead of master and it'll only show the different commits, and they'll even retarget master when the other PR is merged (and of course entirely at your peril when the PRs inevitably get messed up)

**Comment by notspiff** [comment]
@lrusak ping, i've been waiting for almost a week now.

File: xbmc/application/ApplicationPowerHandling.h
Comment (lrusak) [inline xbmc/application/ApplicationPowerHandling.h:36]: I'm not sure what this is for?

**Review by lrusak — APPROVED** [review]
Looks ok. I wonder about the move of `m_renderGUI` to `CApplicationPowerHandling` though. There is probably a better place for it in the future.

File: xbmc/application/ApplicationPowerHandling.h
Comment (notspiff) [reply xbmc/application/ApplicationPowerHandling.h:36]: to give access to protected members which capp used to have access to through inheritance.

**Comment by notspiff** [comment]
the only two uses of setRenderGUI are related to dpms so it's the most appropriate place for the setting, atleast for now.

**Comment by Acidzero2020** [comment]
After this commit I cannot install a addon form zip anymore.
After pressing the button "Install from zip file" comes the warning and after confirming with yes nothing happens.

**Comment by ksooo** [comment]
For me, "Kodi Log Viewer" Add-on does not work anymore. Nothing happens after selecting one of items from ths add-on's selection dialog (show log, show old log, settings). Reverting this commit fixes it, but I have no clue what's causing the issues.

**Comment by ksooo** [comment]
@notspiff AppMessaging is broken:

https://github.com/xbmc/xbmc/blob/master/xbmc/messaging/helpers/DialogHelper.cpp#L39-L45

For the ShowYesNoCustomDialog -> InstallFromZip case, SendMessage returns immediately, with CHOICE_NO, without any user interactions

**Comment by ksooo** [comment]
@enen92 fyi

**Pull request merged**
